x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/MetricAggExtractor.java

```diff
@@ -115,17 +131,17 @@ public Object extract(Bucket bucket) {
         }
 
         Object v = agg.getProperty(property);
-        return handleDateTime(innerKey != null && v instanceof Map ? ((Map<?, ?>) v).get(innerKey) : v);
+        return handleTargetType(innerKey != null && v instanceof Map ? ((Map<?, ?>) v).get(innerKey) : v);
     }
 
-    private Object handleDateTime(Object object) {
-        if (isDateTimeBased) {
-            if (object == null) {
-                return object;
-            } else if (object instanceof Number) {
+    private Object handleTargetType(Object object) {
```

Comment:
Could we have a comment, probably on the method, which says: `This is needed for the SELECT pipeline`, and the casting in the Painless script is for the `HAVING (bucket selector) on the aggregation`?

Reply:
Sure, added.

Reply:
I am not seeing a comment.... was it lost inbetween commits?

Reply:
They've been inlined with other explanatory comments:
https://github.com/elastic/elasticsearch/pull/71525/files#diff-ecb07a105ef132c93f804826078f5f1415929b5f373adc9aa9fe22ae84d855a0R141
https://github.com/elastic/elasticsearch/pull/71525/files#diff-25627339dd1c616298440c7bf7c9c08caa0a681c1df442c004fa71e84bb556a2R128